packages/core/src/interpreter.ts

```diff
@@ -409,16 +406,33 @@ export class Interpreter<
             State.from(initialState, this.machine.context)
           );
 
-    this.scheduler.initialize(() => {
-      this.update(resolvedState, initEvent as SCXML.Event<TEvent>);
+    this._state = resolvedState;
+
+    this.update(resolvedState);
+
+    if (this.options.devTools) {
+      this.attachDevTools();
+    }
+
+    this.flush();
 
-      if (this.options.devTools) {
-        this.attachDevTools();
-      }
-    });
     return this;
   }
 
+  private flush() {
+    this.mailboxStatus = 'processing';
```

Comment:
i wonder - is this status required? couldn't it be just derived from the length of the mailbox?

Reply:
I don't think it can be easily derived because we need to prevent the race condition that sending an event causes another event to be fired, which would flush the mailbox again, even though it's already being flushed. In that case, the mailbox length doesn't matter.